config/custom-convict-formats.ts

```diff
@@ -0,0 +1,99 @@
+import { DbConnection, DbPool } from './knexfile'
+// Needed since connection could be a string or an object
+export const DbConnectionFormat = {
+    name: 'db-connection',
+    validate: function (val: any) {
+        // String check - i.e. SQLite will use a string to specify connection in memory
+        if (typeof val === 'string' || val instanceof String) return true;
+
+        //  Object check
+        if (typeof val === 'object' && val !== null) {
+            const connection = val as DbConnection;
+
+            // Check that object is DbConnection and has DbConnection fields - 
+            // i.e. PG and MySQL use a DbConnection object to configure
+            // Verify that all fields are filled AND they are the right format - presence + format check
+            if (connection.host == null || typeof connection.host !== 'string') {
```

Comment:
```suggestion
            if (!(typeof connection?.host === 'string')) {
```
let apply this compact ?. style also below

Reply:
@eoln now that I think about it, do we even need the optional because here we are saying that connection could be null when in reality, we assert above that connection is not null

Reply:
Maybe we can just use `connection.host !== 'string'`